### ISSUE TITLE
[BUG] Window closes by itself and sometimes gets stuck on sending the first mission(more rarely).

### ISSUE THREAD

**Issue by theruan** [TLDRMissions/TLDRMissions#622]
TLDR version: 10.2-004

After the last update(was actually at version 002 before) the window often closes by itself right after starting to calculate, also, more rarely, the first mission seems to get stuck when it's going to send.
![missions stuck at sending](https://github.com/TLDRMissions/TLDRMissions/assets/36272502/4497818b-80cb-470e-a64e-e76976e4ffb5)
Is there anywhere i can download the version 002 to compare? tried curse, but it was removed from there.

if the autoclose was a feature intended to force the window to close to avoid attempting to send a mission without a proper right( cause blizz UI sometimes doesn't allow it ending up getting stuck on sending) the event GARRISON_MISSION_NPC_OPENED was working fine here, had a weakauras alert set up, when the table is opened and this event isn't fired missions can't be completed or sent, if it was, all good, but now it seems to be trying to close the table too early even though this event was fired and it could complete/send and sometimes gets stuck even though it could.

**Comment by theruan** [comment]
Stuck not so rarely i guess.
![error2](https://github.com/TLDRMissions/TLDRMissions/assets/36272502/81d8f451-2145-4c6e-89fb-2c55f791ccfa)

After further testing, the auto close is related to the option "Open loot window at Mouse"(blizz interface) disabled, sometimes, when this is disable, after the loot window opens and closes the table is forced closed.

This may not have changed since 002, but i did change the setting "Open ..." yesterday, so could always have been like that and be just a blizz bug that can't be avoided.

As for getting stuck, is there a log on what the addon does that i could check? likely too rare to replicate.

On the note of "stucks", sometimes, even more rarely, after pressing the start button, the missions get completed BUT it doesn't start to calculate the missions to be sent, it shows "0" at the start button, pressing the button again restarts the process.

**Comment by teelolws** [comment]
So, for the issue of the windows becoming unresponsive: this is something I've had happen back when I was first making this addon, when I was still gathering data, and wasn't running any addons. The window would just sometimes become unresponsive. I'd try and finish a mission and nothing would happen. I'd try to send one out, and nothing would happen. No errors, it was just like the commands weren't working. It still happens occasionally, and neither the base UI nor my addon will work.

A reload or relog fixes it when it happens.

**Comment by teelolws** [comment]
I am not able to replicate the window closing issue. I need more detail about the problem, what you're doing when it happens, what other addons could be interfering, etc.

**Comment by theruan** [comment]
the window closing issue happened once the "Open loot window at Mouse" was disabled, so it would open at the top left corner(if this is enabled, which i think it is by default, the issue is gone) the only additional thing I have related to this is the addon "FasterLoot", but supposedly it shouldn't affect this, at least in theory it only speeds the auto loot.
The behavior when it happens is pretty much like if u had pressed "Escape" key right after the completion of the missions IF there is a box to be opened/looted(like herbs and such), I use the addon "Open-Sesame" to auto open boxes like those.

If the reason for this is what i think, it may not be able to correct the issue, cause some interfaces force the closing of other windows so they don't overlap, unless modded(like TSM does with the AH interface and profession window).
I will try to manually open stuff with that option disabled tomorrow morning to try to simulate without other addons.

**Comment by teelolws** [comment]
Oh you're opening the cache boxes while trying to keep the garrison window open? Yeah thats not really something I can change as Blizzard controls whether the mission table is visible, not me. I only control the widget to the side of it.

**Comment by theruan** [comment]
yeah, if that option is active(to open at mouse position) the window(tldr/table) isn't closed, so a sugestion would be to check if the option is disabled and warn the user maybe?
on the note of bugs and stuff, checking if the event GARRISON_MISSION_NPC_OPENED was fired before allowing an attempt to complete/send missions would be nice, so the user doesn't keep waiting, the blizz ui(at least for me) doesn't fire this about 10~20% of the time, which can be a pain when u just had to wait 2~5min calculating missions like "break the briarbane" and 1~2 others that can take awhile, as it can't resume the sending only, because i'm guessing the addon was expecting a reply with a confirmation of sending, but the blizz ui isn't accepting the command to send the mission, which forces us close/reopen the window, then cancel and recalculate.

weirdly the error that was happening even with the event being fired didn't happen anymore, at least not in the last 40~50 times i sent missions today, maybe server bug or lag.

**Comment by teelolws** [comment]
Again, I can't fix the window closing issue as thats Blizzards code, not mine.

If you're using the "Allow processing anywhere" option and walking away from the table to calculate the "break the briarbane" mission, then you will need to press the Start button when its ready and you're back at the table. This is by design, you cannot have both "Allow processing anywhere" and "Auto start" selected, so that it doesn't try to send missions when you're not at the table.

**Comment by theruan** [comment]
Sometimes, when we open the table, and decide we want to send missions, and NOT using the "allow processing anywhere", but using the "auto start", the mission table won't allow sending mission because blizzard didn't detect that the person is at the table properly and didn't give the access to the completing/sending mission, so when the addon attempts to send a mission, it can't and gets stuck, this can be fixed by waiting on the event GARRISON_MISSION_NPC_OPENED.

If you filter the events and open/close the mission table a few times you will notice that sometimes this event is not fired, on those times, any attempt to complete a mission or send one will result in being stuck.

**Comment by teelolws** [comment]
How about its counterpart, does `GARRISON_MISSION_NPC_CLOSED` always fire? Also, when it doesn't fire, and the missions won't send, does `C_Garrison.IsAtGarrisonMissionNPC()` return true anyway?

**Comment by theruan** [comment]
C_Garrison.IsAtGarrisonMissionNPC() does return true, and oddly the counterpart in the SL mission table is GARRISON_SHIPYARD_NPC_CLOSED, GARRISON_MISSION_NPC_CLOSED is never fired.
![event](https://github.com/TLDRMissions/TLDRMissions/assets/36272502/3b28c897-2fe8-434d-9a0f-e8ef06c7cc78)

**Comment by teelolws** [comment]
Okay. What I'll probably do is set TLDR Missions to not show when you open the mission table unless it has detected a `GARRISON_MISSION_NPC_OPENED` since login or the last `GARRISON_SHIPYARD_NPC_CLOSED`.

**Comment by teelolws** [comment]
I stood at the mission table for about 5 minute spamming close and open, but did not have a `GARRISON_MISSION_NPC_OPENED` fail to fire. Uploading a fix now, relying on you to tell me if it works. The intention is to move from `ADVENTURE_MAP_OPENED` to `GARRISON_MISSION_NPC_OPENED` when showing my UI.

**Comment by theruan** [comment]
likely will work, what i did myself so far was using a weakauras that did exactly that, it waits on the tldr open event and checks if the 'npc opened' was fired, if it wasn't, it pops a huge ugly icon haha
Based on my tests, if the user already opened the table once, and it worked, the 'npc open' will always fire, it seems to me that this error occurs only in the first minute or MAYBE after a zone change, but i couldn't replicate that right now, but I also remember that a few times after depositing anima i had that error a few times.

[weakauras stuck check.txt](https://github.com/TLDRMissions/TLDRMissions/files/13533636/weakauras.stuck.check.txt)

**Comment by theruan** [comment]
I'm not THAT good at programing yet, BUT If the conditional to check this is:
`local function eventHandler(self, event, ...)
    local arg1 = ...
    if event == "GARRISON_MISSION_NPC_OPENED" then
	print("NPC_OPEN_EVENT DONE")
        if arg1 ~= 123 then return end
        if C_Map.GetBestMapForUnit("player") == 2022 then return end -- this is The Waking Shores. There is an interactable "Scouting Map" that passes in 123 for some reason.
        if C_Map.GetBestMapForUnit("player") == 2024 then return end -- same with Azure Span - from the Blue Dragon quests campaign
        adventureMapOpenHandler(arg1)
        preloadItemRewards()
        if #C_Garrison.GetCompleteMissions(123) == 0 then
            addon.GUI.CompleteMissionsButton:Hide()
        else
            addon.GUI.CompleteMissionsButton:Show()
        end
    elseif (event == "ADDON_LOADED") then`

See my `print("NPC_OPEN_EVENT DONE")` added?
I modified the addon to force print and be able to understand IF the event was indeed fired or if it is passing ahead without it somehow and it indeed is.
In this print the added command "print("NPC_OPEN_EVENT DONE")" didn't execute, my weakauras alert was fired, cause the addon was opened but the event didn't trigger.
![1](https://github.com/TLDRMissions/TLDRMissions/assets/36272502/cb5f7934-9d56-4ace-977c-a11921413cc2)


In this print, I reopened the table, the event was fired and then the print command is shown in chat.
![2](https://github.com/TLDRMissions/TLDRMissions/assets/36272502/a62b2795-8965-4327-8f11-e39cdb671673)


So, is there any other place checking for this event, so i could try to modify, add some prints and debug this?

**Comment by teelolws** [comment]
The only code that should be firing during ADDON_LOADED is the Follower List code.

**Comment by theruan** [comment]
Yeah, i thought at first the code being run was triggered from inside the ADDON_LOADED > Follower List code, cause it was the IF that was valid, but i changed the code myself to include that section(to test only) inside the "garrison open" event(so it would only run if it was 100% valid, then i noticed that even if all 'IFs' in that section are false it does load the table most of the time.

Then i got lost cause i have no clue where to search outside that.
Which file is calls that function first?
so i can try to adds some 'prints' before/after that function.

**Comment by teelolws** [comment]
Its the code at the bottom of the file `EventUtil.ContinueOnAddOnLoaded(addonName, function()`, its there to set up the saved variables once the addon is fully loaded, and I should be able to remove everything after the `    addon:RefreshProfile()`

**Comment by teelolws** [comment]
Eh. If it was still Shadowlands I would, but this addon has declined in popularity enough that I think the few remaining people who still use it will be aware of the problem already.

**Comment by SketchPadd** [comment]
Hi, i had this problem too, installing an older version of this addon solved the problem completely.
